merkledag/merkledag_test.go

```diff
@@ -241,9 +242,10 @@ func TestFetchGraph(t *testing.T) {
 	// create an offline dagstore and ensure all blocks were fetched
 	bs := bserv.New(bsis[1].Blockstore(), offline.Exchange(bsis[1].Blockstore()))
 
-	offline_ds := NewDAGService(bs)
+	// we know the default dagService implements LinkGetter
+	offlineDS := NewDAGService(bs).(ipld.LinkGetter)
```

Comment:
This is exactly the case for having `NewDAGService` return the concrete type.

Reply:
But it is a case only used in tests (and tests in the same module). From an external user's point of view, it's not possible to know what *dagService can do because it's unexported.